### PR TITLE
[SERVICE-458] Prevent applications from receiving their own broadcasts

### PR DESCRIPTION
**Pull request by tim-dinsdale**
So I know that the name is wrong, and you need to tell me what to put there. The rest is AWESOME

### CODE REVIEW

**Review by pjbroadbent — APPROVED** [review]

**Comment by tomrobertsOF** [comment]
Now that the behaviour is no longer TBD, can we fill out implementations for the two tests in `contextLIstener.test.ts` (lines 57 and 94) so we can avoid breaking this in future.

File: test/demo/contextListener.test.ts [hunk 1]
```diff
@@ -91,7 +97,14 @@ describe('Context listeners and broadcasting', () => {
                 }
             });
 
-            test.todo('When calling broadcast from the first app [behavior TBD - do apps receive their own broadcasts?]');
+            test('When calling broadcast from the first app', async () => {
+                await fdc3Remote.broadcast(testAppIdentity, validContext);
+
+                const receivedContexts = await Promise.all(listeners.map(listener => listener.getReceivedContexts()));
+                receivedContexts.forEach((rc)=>console.log(rc));
```
Comment (bryangaleopenfin) [inline test/demo/contextListener.test.ts]: Is this logging intentional or leftover debugging? And if intentional, would put spaces around '=>' for consistency

**Review by bryangaleopenfin — APPROVED** [review]
Approved, but just want to check about the logging

File: test/demo/contextListener.test.ts [hunk 2]
```diff
@@ -54,7 +54,13 @@ describe('Context listeners and broadcasting', () => {
                 expect(receivedContexts[0]).toEqual(validContext);
             });
 
-            test.todo('When broadcast is called from the same app [behavior TBD - do apps receive their own broadcasts?]');
+            test('When broadcast is called from the same app', async () => {
```
Comment (tomrobertsOF) [inline test/demo/contextListener.test.ts]: Just a quick note on the test titles. It looks like you just removed the placeholder part of the name but didn't replace it with anything. If you look at the others we have something along the lines of `When doing this thing, that stuff will happen`, so for this case I would prefer something like `When broadcast is called from the same app the listener is not triggered`

File: test/demo/contextListener.test.ts [hunk 3]
```diff
@@ -91,7 +97,14 @@ describe('Context listeners and broadcasting', () => {
                 }
             });
 
-            test.todo('When calling broadcast from the first app [behavior TBD - do apps receive their own broadcasts?]');
+            test('When calling broadcast from the first app', async () => {
+                await fdc3Remote.broadcast(testAppIdentity, validContext);
+
+                const receivedContexts = await Promise.all(listeners.map(listener => listener.getReceivedContexts()));
+                receivedContexts.forEach((rc)=>console.log(rc));
+                expect(receivedContexts.length).toBe(2);
```
Comment (tomrobertsOF) [inline test/demo/contextListener.test.ts]: This length check is covered by the equals check below it, so not really necessary.

**Review by rypete — APPROVED** [review]
Looks good

**Review by tomrobertsOF — APPROVED** [review]

**Comment by tomrobertsOF** [comment]
Passed on CI, so going to go ahead and merge.

**Pull request merged**